### PR TITLE
Return cors headers on error pages, submit queue

### PR DESCRIPTION
**Pull request by alastair**
The scraper xhr request to listenbrainz fails on http 4xx and 5xx error pages because the response doesn't include any cors headers.
We should set these headers on the request and check the status code in the javascript on the "success" callback.
We should not retry if the error is 4xx (bad data, bad auth), but retry if it's 5xx (possibly a transient server error)

Still to be done:

 - [x] implement 401 handler
 - [x] retry on the right error codes in the scraper 
 - [x] refactor error handlers to use a helper function

This is http://tickets.musicbrainz.org/browse/LB-21

### CODE REVIEW

**Comment by alastair** [comment]
also added submit queue: http://tickets.musicbrainz.org/browse/LB-40

File: webserver/templates/user/scraper.js [hunk 1]
```diff
@@ -121,17 +121,19 @@ function getLastFMPage(page, callback) {
...
 var numCompleted = 0;
+var activeSubmissions = 0;
 
 function dispatch() {
-    for (var i = 0; i < toReport.length; ++i) {
+    for (var i = 0; i < toReport.length; i++) {
         reportScrobbles(toReport[i]);
     }
-    toReport = [];
+    toReport = []
```
Comment (gentlecat) [inline webserver/templates/user/scraper.js]: Don't need a semicolon there?

Comment (alastair) [inline webserver/templates/user/scraper.js]: that's what I get for rewriting a block and not checking diff

File: webserver/templates/user/scraper.js [hunk 2]
```diff
@@ -121,17 +121,19 @@ function getLastFMPage(page, callback) {
...
+var version = "1.2";
 var page = 1;
 var numberOfPages = parseInt(document.getElementsByClassName("pages")[0].innerHTML.trim().split(" ")[3]);
 
 var toReport = [];
 var numCompleted = 0;
+var activeSubmissions = 0;
 
 function dispatch() {
-    for (var i = 0; i < toReport.length; ++i) {
```
Comment (mayhem) [inline webserver/templates/user/scraper.js]: What do you have against pre-decrement?? Hater!

**Comment by mayhem** [comment]
Seems ok, with my limited JS knowledge.

**Comment by alastair** [comment]
I tested this by making the server return 200, 400, 401, 500 and checking the logs to see that it did the right thing

**Pull request merged**